### PR TITLE
net-dialup/radiusclient: Fix cross compilation by letting user manage…

### PR DESCRIPTION
**Pull request by bjacquin**
… ac_cv_struct_utsname and ac_cv_file__dev_urandom, bug #373347

Package-Manager: portage-2.2.20.1

### CODE REVIEW

**Comment by monsieurp** [comment]
@Pinkbyte please review.

**Comment by mgorny** [comment]
@bjacquin, commit message length again ;-).

File: net-dialup/radiusclient/files/radiusclient-0.3.2-BJA-cross.diff [hunk 1]
```diff
@@ -0,0 +1,67 @@
+--- configure.in	2002-02-03 14:51:40.000000000 +0100
```
Comment (mgorny) [inline net-dialup/radiusclient/files/radiusclient-0.3.2-BJA-cross.diff]: Where does this bug come from? It is a good practice to leave some notes at top of the patch, like what it does and where it comes from, if it's committed upstream, if it has upstream bug etc.

**Comment by bjacquin** [comment]
Pull request updated as requested

File: net-dialup/radiusclient/files/radiusclient-0.3.2-BJA-cross.diff [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+ AC_CHECK_FUNCS(flock fcntl uname gethostname sysinfo getdomainname)
+ AC_CHECK_FUNCS(stricmp random rand snprintf vsnprintf)
+ 
+-if test "$ac_cv_func_uname" = 'yes'
+-then
+-	AC_MSG_CHECKING([for field domainname in struct utsname])
+-	AC_TRY_RUN([
+-	#include <sys/utsname.h>
++AC_CACHE_CHECK([for field domainname in struct utsname], ac_cv_struct_utsname,
++ [AC_TRY_RUN([
```
Comment (mgorny) [inline net-dialup/radiusclient/files/radiusclient-0.3.2-BJA-cross.diff]: The core issue here is using `AC_TRY_RUN` unnecessarily. `AC_TRY_COMPILE` should be enough.

**Comment by mgorny** [comment]
It would be easier to upstream this if you wrote separate patches for each block of code being fixed. And you really should submit this upstream ;-).

File: net-dialup/radiusclient/files/radiusclient-0.3.2-BJA-cross.diff (hunk 1 above)
Comment (bjacquin) [inline net-dialup/radiusclient/files/radiusclient-0.3.2-BJA-cross.diff]: The patch comes from me as metionned in the pach head. Patch was reported upstream on 28/06/2011 15:22 http://freecode.com/projects/radiusclient/tickets/2c46a43c4ab what now leads to a 404 error. Looking at sending it back

File: net-dialup/radiusclient/files/radiusclient-0.3.2-BJA-cross.diff (hunk 2 above)
Comment (bjacquin) [inline net-dialup/radiusclient/files/radiusclient-0.3.2-BJA-cross.diff]: AC_TRY_COMPILE as no cross-compile fallback unfortunately

File: net-dialup/radiusclient/files/radiusclient-0.3.2-BJA-cross.diff (hunk 1 above)
Comment (mgorny) [inline net-dialup/radiusclient/files/radiusclient-0.3.2-BJA-cross.diff]: Aye, freecode is basically dead. I hope the project has some upstream these days…

File: net-dialup/radiusclient/files/radiusclient-0.3.2-BJA-cross.diff (hunk 2 above)
Comment (mgorny) [inline net-dialup/radiusclient/files/radiusclient-0.3.2-BJA-cross.diff]: It doesn't have to. As the name 'cross-compiling' suggests, you can compile there :-P.

**Comment by bjacquin** [comment]
Author contact to see if he still accept patches as it looks like the project did not updated since 2002. Project might be abandoned

File: net-dialup/radiusclient/files/radiusclient-0.3.2-BJA-cross.diff (hunk 2 above)
Comment (bjacquin) [inline net-dialup/radiusclient/files/radiusclient-0.3.2-BJA-cross.diff]: Then the following error is triggered when compiling on target for target:

checking for field domainname in struct utsname... configure: error: cross-compiling,
  presetting ac_cv_struct_utsname=(yes|no) will help

File: net-dialup/radiusclient/files/radiusclient-0.3.2-BJA-cross.diff (hunk 1 above)
Comment (bjacquin) [inline net-dialup/radiusclient/files/radiusclient-0.3.2-BJA-cross.diff]: Maintainer contacted on a side channel

File: net-dialup/radiusclient/files/radiusclient-0.3.2-BJA-cross.diff (hunk 2 above)
Comment (mgorny) [inline net-dialup/radiusclient/files/radiusclient-0.3.2-BJA-cross.diff]: Are you sure you are using `AC_TRY_COMPILE` correctly? Note that the arguments are different than for `AC_TRY_RUN`.

```
 -- Macro: AC_TRY_COMPILE (INCLUDES, FUNCTION-BODY, [ACTION-IF-FOUND [,
          ACTION-IF-NOT-FOUND]])

 -- Macro: AC_TRY_RUN (PROGRAM, [ACTION-IF-TRUE [, ACTION-IF-FALSE [,
          ACTION-IF-CROSS-COMPILING]]])

Comment (bjacquin) [inline net-dialup/radiusclient/files/radiusclient-0.3.2-BJA-cross.diff]: You are right, I messed an argument

**Comment by bjacquin** [comment]
Here is the response from the author:

> the old radiusclient code seems to be part of FreeRADIUS nowadays.  Please see
> http://wiki.freeradius.org/project/Radiusclient.  Current version is 1.1.7
> dated 01/19/2015.

So, it might be more relevant to remove the package from Gentoo ?

**Comment by mgorny** [comment]
Sounds good to me, going to contact maintainer about this.

**Comment by mgorny** [comment]
By the way, could you see if `radclient` from freeradius is the same thing as radiusclient here? I'd like to know if we need pure removal or if we need to package something.

**Comment by Pinkbyte** [comment]
API in this lib is old and Exim still relies on it. I have tried to make it build with either net-dialup/freeradius-client or net-dialup/radiusclient-ng - no luck :-(

**Comment by mgorny** [comment]
@Pinkbyte, can I merge this PR then?

**Comment by Pinkbyte** [comment]
@mgorny, there is some progress in fixing of consumers to use net-dialup/freeradius-client, thanks to Alexander Tsoy(@puleglot), who is also reported these issues upstream

See https://bugs.gentoo.org/show_bug.cgi?id=563410 and https://bugs.gentoo.org/show_bug.cgi?id=563356

As net-dialup team member I have no objections on merging this if you want it fixed, but future of this package is definitely one - be lastrited

**Comment by mgorny** [comment]
Ok, let's focus on removing it then. Thanks for the work, everyone.

**Pull request closed without merging**